pkg/agent/agent.go

```diff
@@ -409,10 +424,11 @@ func (i *Initializer) configureGatewayInterface(gatewayIface *interfacestore.Int
 		return err
 	}
 
+	i.nodeConfig.GatewayConfig = &config.GatewayConfig{Name: i.hostGateway, MAC: gwMAC}
```

Comment:
should we also set LinkIndex here since it becomes available at the same time as gwMAC? then the following code deals with IP only.

Reply:
Ping @jianjuns, just in case you missed this comment.

Reply:
Right, but here I am just keeping the previous behavior that does not set LinkIndex for NetworkPolicyOnly mode. I think ok to keep that behavior too.